pkg/neorpc/filters.go

```diff
@@ -49,6 +53,12 @@ type (
 	}
 )
 
+// SubscriptionFilter is an interface for all subscription filters.
+// IsValid checks whether the filter is valid.
+type SubscriptionFilter interface {
+	IsValid() error
+}
+
```

Comment:
Create `var ErrInvalidSubscriptionFilter = errors.New("invalid subscription filter")` here with appropriate comment and return it as a wrapped error everywhere.